apps/meteor/app/api/server/v1/rooms.ts

```diff
@@ -860,6 +862,45 @@ API.v1.addRoute(
 	},
 );
 
+API.v1.addRoute(
+	'rooms.membersOrderedByRole',
+	{ authRequired: true, validateParams: isRoomsMembersOrderedByRoleProps },
+	{
+		async get() {
+			const findResult = await findRoomByIdOrName({
+				params: this.queryParams,
+				checkedArchived: false,
+			});
```

Comment:
Correct me, but if this `findRoomByIdOrName` call doesn't check for the user's permissions or access, then this will leak the list of members of private rooms to non admins, right? 👀

Reply:
👀  Yes you are correct. I have added the canAccessRoomAsync logic in the endpoint to check for the user permissions.